collector/os.go

```diff
@@ -261,12 +257,21 @@ func (c *OSCollector) collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) (
 		timezoneName,
 	)
 
-	ch <- prometheus.MustNewConstMetric(
-		c.PagingFreeBytes,
-		prometheus.GaugeValue,
-		pfb,
-	)
-
+	if pagingErr == nil {
```

Comment:
Could this be handled closer to the error return possibility 80 lines higher? There's been another, different registry access since; I would find it easier to follow the flow with less to read.

(Obviously, I'm also random person on the internet; I hope review comments are welcome. I'm coming from #923 where I tried to understand an error from near here and haven't yet)

Reply:
I agree that the flow of the collector is made more difficult by the change, but the `pagingFiles` variable is required shortly afterwards to populate the `fsipf` and `pfb` variables, while the metric exports to the channel all occur at the same time to improve readability.

So I don't think it's worth trying to rearrange the `collect` function to group `pagingErr` with the paging metrics.